xmppserver/src/main/java/org/jivesoftware/openfire/server/ServerDialback.java

```diff
@@ -643,6 +643,10 @@ private boolean isHostUnknown(String recipient) {
         if (host_unknown && recipient.contains(domainPair.getLocal())) {
             host_unknown = !routingTable.hasComponentRoute(new JID(recipient));
         }
+
+        if (host_unknown) {
+            host_unknown = Trunking.isTrunkingEnabledFor(recipient);
+        }
```

Comment:
I think this logic got inverted when the switch to SystemProperties happened

```suggestion
        if (host_unknown) {
            host_unknown = !Trunking.isTrunkingEnabledFor(recipient);
        }
```